### PR TITLE
depend-o-pocalipse

### PR DESCRIPTION
**Pull request by gilescope**
Fortune favors the brave. 

There seem to be a few dependencies that I can get it to compile without it needing them.

Todo:
- [x] unused dependencies removed
- [x] unused dev dependencies included.
- [x] green build
- [x] check cumulus build
- [x] check frame-benchmarking
- [x] check try-runtime
- [x] mention on community channel

### CODE REVIEW

File: bin/node/bench/Cargo.toml [hunk 1]
```diff
@@ -11,9 +11,9 @@ license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 [dependencies]
 log = "0.4.8"
 node-primitives = { version = "2.0.0", path = "../primitives" }
-node-testing = { version = "3.0.0-dev", path = "../testing" }
-node-runtime = { version = "3.0.0-dev", path = "../runtime" }
-sc-cli = { version = "0.10.0-dev", path = "../../../client/cli" }
+node-testing = { version = "4.0.0-dev", path = "../testing" }
+node-runtime = { version = "4.0.0-dev", path = "../runtime" }
+#sc-cli = { version = "0.10.0-dev", path = "../../../client/cli" }
```
Comment (expenses) [inline bin/node/bench/Cargo.toml:16]: I'd prefer that we simply removed these lines instead of commenting them out.

Comment (gilescope) [reply bin/node/bench/Cargo.toml:16]: absolutely. once I get a green build :-)

Comment (gilescope) [reply bin/node/bench/Cargo.toml]: got a green build so they are gone.

**Comment by shawntabrizi** [comment]
How do you identify these? Some tool/script?

**Comment by gilescope** [comment]
Yup, cargo metadata on the workspace, then cargo edit's cargo rm on each dep and cargo check --all-targets to see if it built afterwards. record good results to a file and then git reset --hard and try a different dep.

It ran all night but it seemed to not do too bad a job.... The down side is I don't have enough color to know if there's any reason we can't remove these dependencies. This is where I need your help to point out anything deeply suspicious - "surely that crate's used by x" etc.

I'm going to have a follow up PR that does the same but for dev-dependencies but I want to land them separately.
(We can brush up the function into something nice and put it into cargo unleashed if it is useful to people - there's already a subcommand in cargo-unleash that does something in this direction but isn't quite as brute force.)

**Review by gnunicorn — APPROVED** [review]
Neat. Nothing too suspicious coming up glancing over them, there's a good chance these are indeed not needed (anymore). 

But two questions on your process, just to make sure we don't accidentally dropped one too many:
1. did you build in `release`-mode(, too)? 
2. have the  frame/pallets been build for wasm/`no_std`?

**Comment by gilescope** [comment]
Both good questions. I did build a couple of pallets - is there a
Cargo.toml with all the no_std crates in as a workspace? Or will all frame
& pallets cover it? I was assuming CI tested release mode as well as wasm -
there were certainly some wasm ci errors initially.

Will run a few more checks and will report back here.

**Comment by gilescope** [comment]
Release build is clean. I'm unclear at the moment how to build just the wasm for the whole workspace.

**Comment by thiolliere** [comment]
The pallet/frame crate should be good if a release build of node-runtime is done.

**Review by thiolliere — APPROVED** [review]

File: primitives/consensus/common/Cargo.toml [hunk 2]
```diff
@@ -15,22 +15,24 @@ targets = ["x86_64-unknown-linux-gnu"]
...
 futures-timer = "3.0.1"
 sp-std = { version = "4.0.0-dev", path = "../../std" }
 sp-version = { version = "4.0.0-dev", path = "../../version" }
 sp-runtime = { version = "4.0.0-dev", path = "../../runtime" }
 sp-utils = { version = "4.0.0-dev", path = "../../utils" }
-sp-trie = { version = "4.0.0-dev", path = "../../trie" }
 sp-api = { version = "4.0.0-dev", path = "../../api" }
-parking_lot = "0.11.1"
+#codec = { package = "parity-scale-codec", version = "2.0.0", features = ["derive"] }
+#parking_lot = "0.11.1"
```
Comment (bkchr) [inline primitives/consensus/common/Cargo.toml:33]:
```suggestion
```

File: frame/contracts/Cargo.toml [hunk 3]
```diff
@@ -40,7 +45,7 @@ sp-std = { version = "4.0.0-dev", default-features = false, path = "../../primit
 [dev-dependencies]
 assert_matches = "1"
 hex-literal = "0.3"
-paste = "1"
+# paste = "1"
```
Comment (athei) [inline frame/contracts/Cargo.toml:48]:
```suggestion
```

Comment (gilescope) [reply frame/contracts/Cargo.toml:48]: yep, will remove these once I get a green build that includes the removal of the dev dependencies.

**Comment by athei** [comment]
> Yup, cargo metadata on the workspace, then cargo edit's cargo rm on each dep and cargo check --all-targets to see if it built afterwards. record good results to a file and then git reset --hard and try a different dep.

I used [cargo outdatd](https://crates.io/crates/cargo-outdated/0.4.0) in the past to identify unused dependencies. You did try that?

**Comment by athei** [comment]
> Yup, cargo metadata on the workspace, then cargo edit's cargo rm on each dep and cargo check --all-targets to see if it built afterwards. record good results to a file and then git reset --hard and try a different dep.

Did you try [cargo-udeps](https://crates.io/crates/cargo-udeps)?

File: frame/contracts/Cargo.toml (hunk 3 above)
Comment (gilescope) [reply frame/contracts/Cargo.toml]: removed now we're getting close to the finish line.

**Comment by gilescope** [comment]
(There's some auto-formatting of the toml files going on here which is nice, though unclear how vscode is managing this?)

File: client/tracing/src/logging/event_format.rs [hunk 4]
```diff
@@ -93,8 +93,7 @@ where
 
 		// Custom code to display node name
 		if let Some(span) = ctx.lookup_current() {
-			let parents = span.parents();
-			for span in std::iter::once(span).chain(parents) {
+			for span in span.scope() {
```
Comment (gilescope) [inline client/tracing/src/logging/event_format.rs:96]: Slight tidy up here: scope() includes all parents including self. 
(Fixes a deprecation warning)

**Comment by gilescope** [comment]
Aside from the above no other `.rs` has been changed - the others are just cargo fmt. (Potentially we should pin the compiler via a rust-toolchain.toml so that we manage when we take these rustfmt changes)

File: bin/node/browser-testing/Cargo.toml [hunk 5]
```diff
@@ -1,24 +1,20 @@
 [package]
 name = "node-browser-testing"
-version = "3.0.0-dev"
+version = "4.0.0-dev"
```
Comment (athei) [inline bin/node/browser-testing/Cargo.toml:3]: Why are major versions bumped?

Comment (gilescope) [reply bin/node/browser-testing/Cargo.toml:3]: Yeah maybe that's my bad. I was aiming for consistency of everything being either v 4.0.0 or 0.10.0. but talking with Ben it seems that we're heading towards every crate having its own independent version number.

**Review by athei — APPROVED** [review]

**Pull request merged**